### PR TITLE
Add record and valuetype for checkpoint

### LINKED ISSUE #9719: Add records for checkpoint
Depends on #9612 

Goal:
Define required types in protocol and create classes for the following records (decide proper names):

- checkpoint:create (command)
   - consists of:
      -   checkpoint id : long
- checkpoint:created (event)
     -  consists of:
        -  checkpoint id
        -  checkpoint position
- checkpoint:ignored (event)

Context:
`StreamProcessor` triggers backup when processing `checkpoint:create` command.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Checkpoints are the basis for the backups. Each backup consists of a consistent checkpoint of partitions.

We are adding the new checkpoint record together with the bpmn records. This is because:
 -  We want to export this record for observability. Adding it as a new ValueType lets us reuse the exporter api. Since we want to export it together anyway, distinguishing these different types of records in the protocol is not valuable.
 - Currently protocol is shared by engine and the platform. So we don't get much value by isolating the non-engine ValueTypes from the engine ValueTypes.

## Related issues

closes #9719 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: protocol/src/main/resources/protocol.xml [hunk 1]
```diff
@@ -42,6 +42,9 @@
       <validValue name="DECISION">25</validValue>
       <validValue name="DECISION_REQUIREMENTS">26</validValue>
       <validValue name="DECISION_EVALUATION">27</validValue>
+
+      <!-- Management records / record not related to process automation -->
+      <validValue name="CHECKPOINT">100</validValue>
```
Comment (oleschoenburg) [inline protocol/src/main/resources/protocol.xml:47]: :wrench: how about starting from the bottom, so 255 (or 254?). That way we can leave the some more room for the process automation related records.

File: exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchExporterConfiguration.java [hunk 2]
```diff
@@ -133,6 +135,8 @@ public static class IndexConfiguration {
     public boolean variable = true;
     public boolean variableDocument = true;
 
+    public boolean checkpoint = false;
```
Comment (oleschoenburg) [inline exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchExporterConfiguration.java:138]: :x: this needs to be added to the `toString` method.

**Review by lenaschoenburg — APPROVED** [review]
Awesome! This basically looks good to me, except one adjustment to a `toString`. Keep in mind that I don't have a good mental model of all the places where we are using records so I wouldn't notice if the new record is missing somewhere.

File: protocol/src/main/resources/protocol.xml (hunk 1 above)
Comment (deepthidevaki) [reply protocol/src/main/resources/protocol.xml:47]: :+1:

Comment (deepthidevaki) [reply protocol/src/main/resources/protocol.xml:47]: 255 is used as null/unknown value. So will start with 254.

**Pull request merged**